#------------------------------ tabstop = 4 ----------------------------------
#
# Copyright (C) 2024 Comcast
#
# All rights reserved.
#
# This software is protected by copyright laws of the United States
# and of foreign countries. This material may also be protected by
# patent laws of the United States and of foreign countries.
#
# This software is furnished under a license agreement and/or a
# nondisclosure agreement and may only be used or copied in accordance
# with the terms of those agreements.
#
# The mere transfer of this software does not imply any licenses of trade
# secrets, proprietary technology, copyrights, patents, trademarks, or
# any other form of intellectual property whatsoever.
#
# Comcast retains all ownership rights.
#
#------------------------------ tabstop = 4 ----------------------------------

#
# Created by Kevin Funderburg on 09/11/2024
#

From 9e648458bfa45479de18477e58b6d73cc513b781 Mon Sep 17 00:00:00 2001
From: Jesse Crews <jesse_crews@comcast.com>
Date: Mon, 17 May 2021 10:27:33 -0500
Subject: [PATCH 3/3] Linux: Check for SYS_getdents64 and fallback

 * Define SYS_getdents64 as __NR_getdents64 as required

Signed-off-by: Jesse Crews <jesse_crews@comcast.com>
---
 glib/gspawn.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/glib/gspawn.c b/glib/gspawn.c
index c8e0ca806..08f6affca 100644
--- a/glib/gspawn.c
+++ b/glib/gspawn.c
@@ -1174,6 +1174,14 @@ struct linux_dirent64
   char           d_name[]; /* Filename (null-terminated) */
 };

+#ifndef SYS_getdents64
+    #ifdef __NR_getdents64
+        #define SYS_getdents64 __NR_getdents64
+    #else
+       #error "no dirent64 support (requires Linux 2.4+)"
+    #endif
+#endif
+
 /* This function is called between fork() and exec() and hence must be
  * async-signal-safe (see signal-safety(7)). */
 static gint
--
2.27.0

